Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UCIRangeClauseFilter.cs

```diff
@@ -0,0 +1,932 @@
+                    _ranges.Remove(endIsWithin.First());
+                    _ranges.Add(new Tuple<T, T>(start, original.Item2));
+                    rangeIsAdded = true;
+                }
+                else
+                {
+                    var original = startIsWithin.First();
+                    _ranges.Remove(startIsWithin.First());
+                    _ranges.Add(new Tuple<T, T>(original.Item1, end));
+                    rangeIsAdded = true;
+                }
+            }
+
+            return rangeIsAdded;
+        }
+
+        private void RemoveIsClauseImpl(string opSymbol)
+        {
+            if (_isClause.Keys.Contains(opSymbol))
+            {
+                if (_hasExtents)
+                {
+                    _isClause.Remove(opSymbol);
+                    var extentVal = opSymbol.Equals(CompareTokens.LT) ? _minExtent : _maxExtent;
+                    AddIsClauseImpl(extentVal, opSymbol);
+                }
+                else
+                {
+                    _isClause.Remove(opSymbol);
+                }
+            }
+        }
+
+        private string GetSinglesDescriptor()
+        {
+            var singles = _singleValues.Select(sv => sv.ToString()).ToList();
+            singles.AddRange(_variableSingles);
+            return GetSingleValueTypeDescriptor(singles, "Single=");
+        }
+
+        private string GetRelOpDescriptor()
+        {
+            return GetSingleValueTypeDescriptor(_relationalOps.ToList(), "RelOp=");
+        }
+
+        private string GetSingleValueTypeDescriptor<K>(List<K> values, string prefix)
+        {
+            if (!values.Any()){ return string.Empty; }
+
+            StringBuilder series = new StringBuilder();
+            foreach (var val in values)
+            {
+                series.Append($"{val},");
+            }
+            return $"{prefix}{series.ToString().Substring(0, series.Length - 1)}";
+        }
+
+        private string GetRangesDescriptor()
+        {
+            if(!(_ranges.Any() || _variableRanges.Any())) { return string.Empty; }
+
+            StringBuilder series = new StringBuilder();
+            foreach (var val in _ranges)
+            {
+                series.Append($"{val.Item1}:{val.Item2},");
+            }
+            foreach (var val in _variableRanges)
+            {
+                series.Append(val.ToString() + ",");
+            }
+            return $"Range={series.ToString().Substring(0, series.Length - 1)}";
+        }
+
+        private string GetIsClausesDescriptor(string opSymbol)
+        {
+            var result = string.Empty;
+            if (_isClause.TryGetValue(opSymbol, out List<T> values))
+            {
+                var isLT = opSymbol.Equals(CompareTokens.LT);
+                var value = isLT ? values.Max() : values.Min();
+                var extentToCompare = isLT ? _minExtent : _maxExtent;
+                var prefix = isLT ? "IsLT=" : "IsGT=";
```

Comment:
It's not obvious from this code but is this ultimately going to be user-facing? If so, then we need to make use of `RubberduckUI` to support i18n for the operator description. (I don't think we need to be concerned with other parts since we all use English keywords for data types but we would need to handle `IsLT` or `IsGT` which might be meaningless in a different language, or at least change to `Is <` / `Is >` to correspond with VBA keywords.